plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java

```diff
@@ -295,31 +323,89 @@ private void doVacuum(
+            catch (IOException e) {
+                if (isVacuumLoggingEnabled) {
+                    // This mimics Delta behaviour where it sets metrics to 0 in case of a failure
+                    logVacuumEnd(handle.location(), session, readVersion, 0, 0, "FAILED");
+                }
+                throw e;
+            }
+            if (isVacuumLoggingEnabled) {
+                logVacuumEnd(handle.location(), session, readVersion, filesToDelete.size(), vacuumedDirectories.size(), "COMPLETED");
+            }
             log.info(
                     "[%s] finished vacuuming table %s [%s]: files checked: %s; metadata files: %s; retained known files: %s; retained unknown files: %s; removed files: %s",
                     queryId,
                     tableName,
-                    tableLocation,
+                    tableLocationString,
                     allPathsChecked,
                     transactionLogFiles,
                     retainedKnownFiles,
                     retainedUnknownFiles,
-                    removedFiles);
+                    totalFilesToDelete);
         }
     }
+
+    private void logVacuumStart(String location, ConnectorSession session, long readVersion, long numFilesToDelete, long filesToDeleteSize)
+            throws IOException
+    {
+        long createdTime = System.currentTimeMillis();
+        long commitVersion = readVersion + 1;
+
+        TransactionLogWriter transactionLogWriter = transactionLogWriterFactory.newWriterWithoutTransactionIsolation(session, location);
+        transactionLogWriter.appendCommitInfoEntry(getCommitInfoEntry(commitVersion, createdTime, session, "VACUUM START", ImmutableMap.of("queryId", session.getQueryId()), ImmutableMap.of("numFilesToDelete", String.valueOf(numFilesToDelete), "sizeOfDataToDelete", String.valueOf(filesToDeleteSize)), readVersion));
+
+        transactionLogWriter.flush();
+    }
+
+    private void logVacuumEnd(String location, ConnectorSession session, long readVersion, int numDeletedFiles, int numVacuumedDirectories, String status)
+            throws IOException
+    {
+        long createdTime = System.currentTimeMillis();
+        long commitVersion = readVersion + 2;
+
+        TransactionLogWriter transactionLogWriter = transactionLogWriterFactory.newWriterWithoutTransactionIsolation(session, location);
+        transactionLogWriter.appendCommitInfoEntry(getCommitInfoEntry(commitVersion, createdTime, session, "VACUUM END", ImmutableMap.of("queryId", session.getQueryId(), "status", status), ImmutableMap.of("numDeletedFiles", String.valueOf(numDeletedFiles), "numVacuumedDirectories", String.valueOf(numVacuumedDirectories)), readVersion));
+        transactionLogWriter.flush();
+    }
+
+    private CommitInfoEntry getCommitInfoEntry(long commitVersion, long createdTime, ConnectorSession session, String operation, ImmutableMap<String, String> operationParameters, ImmutableMap<String, String> operationMetrics, long readVersion)
```

Comment:
We usually avoid using `ImmutableMap` for arguments. Please use `Map` instead. 
Also, this line is too long. Please put parameters on separate lines.